### ISSUE TITLE
ValueError in shuffle_tfrecords_beam

### ISSUE THREAD

**Issue by koido** [google/deepvariant#133]
Hi,

I'm Masaru in Japan.
I also asked in https://github.com/google/deepvariant/issues/127 ; but this is a different type of issue.

I also want to train DeepVariant model from the WGS data in our laboratory using our standalone computational resources.
I referred and followed https://github.com/google/deepvariant/blob/r0.7/docs/deepvariant-tpu-training-case-study.md .

First, I successfully made tfrecord file using DeepVariant v0.7.1 (docker).

Then I tried shuffling tfrecords as follows:

    #!/bin/sh
    
    INPUT_PATTERN_LIST=examples.train/sample_id/1/sample_id.tfrecord-?????-of-00056.gz
    for CHROM in `seq 2 19`
    do
     INPUT_PATTERN_LIST="$INPUT_PATTERN_LIST,examples.train/sample_id/$CHROM/sample_id.tfrecord-?????-of-00056.gz"
    done
    
    /usr/bin/python ../../git/deepvariant-r0.7/tools/shuffle_tfrecords_beam.py \
     --input_pattern_list=$INPUT_PATTERN_LIST \
     --output_pattern_prefix=training_set.with_label.shuffled \
     --output_dataset_config_pbtxt=training_set.dataset_config.pbtxt \
     --output_dataset_name=sample_id \
     --runner=DirectRunner > log/shuffle.train.log 2>&1

This script made errors [(please see this file)](https://github.com/google/deepvariant/files/2708579/shuffle.train.0_id_masked.log).

However, the codes which used chromosome-divided tfrecords (chr1-10 and chr11-chr19) worked fine as follows:

    INPUT_PATTERN_LIST=examples.train/sample_id/1/sample_id.tfrecord-?????-of-00056.gz
    for CHROM in `seq 2 10`
    do
     INPUT_PATTERN_LIST="$INPUT_PATTERN_LIST,examples.train/sample_id/$CHROM/sample_id.tfrecord-?????-of-00056.gz"
    done

    /usr/bin/python ../../git/deepvariant-r0.7/tools/shuffle_tfrecords_beam.py \
     --input_pattern_list=$INPUT_PATTERN_LIST \
     --output_pattern_prefix=training_set.with_label.shuffled \
     --output_dataset_config_pbtxt=training_set.dataset_config.pbtxt \
     --output_dataset_name=sample_id \
     --runner=DirectRunner > log/shuffle.train.log 2>&1

and

    INPUT_PATTERN_LIST=examples.train/sample_id/11/sample_id.tfrecord-?????-of-00056.gz
    for CHROM in `seq 12 19`
    do
     INPUT_PATTERN_LIST="$INPUT_PATTERN_LIST,examples.train/sample_id/$CHROM/sample_id.tfrecord-?????-of-00056.gz"
    done

    /usr/bin/python ../../git/deepvariant-r0.7/tools/shuffle_tfrecords_beam.py \
     --input_pattern_list=$INPUT_PATTERN_LIST \
     --output_pattern_prefix=training_set.with_label.shuffled \
     --output_dataset_config_pbtxt=training_set.dataset_config.pbtxt \
     --output_dataset_name=sample_id \
     --runner=DirectRunner > log/shuffle.train.log 2>&1

I cannot figure out the reason why this divided approach succeeded.

Could you tell me any suggestions of points to check?
Because I'm using personal data under strict data management, I cannot provide or share our tfrecords.

I used apache-beam v2.9.0 in python 2.7.5 (not in docker of DeepVariant v0.7.1).
If some version dependencies exist, I'd like to get some dockerfile like your environment.

Best regards,

Masaru

**Comment by sidharthgoel** [comment]
Hi Masaru,
It looks like you're using DirectRunner, which is fine for smaller datasets, but when we have larger datasets we instead use DataflowRunner (see where we shuffle training set [here](https://github.com/google/deepvariant/blob/r0.7/docs/deepvariant-tpu-training-case-study.md#shuffle-each-set-of-examples-and-generate-a-data-configuration-file-for-each)). In the Case Study, the DirectRunner is used for the small validation set and DataflowRunner is used for the large training set.

The fact that it works when you split up the data into smaller pieces suggests this may be the issue. Please try running with DataflowRunner and let me know if that works for you.

**Comment by koido** [comment]
Hi Sidharth,

Thanks for your kind comments.
I understood what is wrong.

However, this page (https://beam.apache.org/documentation/) says:
> DataflowRunner: Runs on Google Cloud Dataflow, a fully managed service within Google Cloud Platform.
Then, are your scripts for shuffling training set available only in Google Cloud Platform?

Best,

Masaru

**Comment by pichuan** [comment]
Hi,
On the same page , it also mentioned other runners such as SparkRunner.
Our shuffling script uses Apache Beam, which can run on different
infrastructures such as Spark or Dataflow, and is not only restricted to
Google Cloud. However, our example was done on Google Cloud. You can set it
up on different things to run.

If you're running on smaller amount of examples where your machine can
handle, it is possible to use DirectRunner.


(from my phone)

**Comment by koido** [comment]
Hi,

I understood what I should do.
Thanks a lot!

Masaru
